### PR TITLE
feat: Adds email signature form to profile settings

### LINKED ISSUE #3378: Feature request: Email signature
It would be great if each agent could set in their **profile settings** a signature that would be automatically added to any outgoing emails (from email Inbox, but also conversation continuity, chat transcript).

There would be a text field that would allow HTML tags, so we could have a signature with logo image, bold text, etc.

I love Chatwoot, but this is the last missing feature preventing me from using it in Production.

### PR DESCRIPTION
**Pull request by nithindavid**
Fixes #3378 
## Type of change
image

image

### CODE REVIEW

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue [hunk 1]
```diff
@@ -556,6 +605,29 @@ export default {
   margin-bottom: 0;
 }
 
+.message-signature-wrap {
+  margin: 0 var(--space-normal);
+  padding: var(--space-small);
+  display: flex;
+  align-items: center;
```
Comment (fayazara) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:612]:
```suggestion
  align-items: baseline;
```

Looks a little better.
CleanShot 2022-02-07 at 11 49 11

Comment (fayazara) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:612]:
```suggestion
  align-items: baseline;
```

Looks a little better.
CleanShot 2022-02-07 at 11 49 11

Comment (fayazara) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:612]:
```suggestion
  align-items: baseline;
```

Looks a little better.
CleanShot 2022-02-07 at 11 49 11

Comment (fayazara) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:612]:
```suggestion
  align-items: baseline;
```

Looks a little better.
CleanShot 2022-02-07 at 11 49 11

File: app/javascript/dashboard/api/auth.js [hunk 2]
```diff
@@ -142,9 +142,9 @@ export default {
   }) {
     const formData = new FormData();
     Object.keys(profileAttributes).forEach(key => {
-      const value = profileAttributes[key];
-      if (value) {
-        formData.append(`profile[${key}]`, value);
+      const hasValue = profileAttributes[key] === undefined;
```
Comment (pranavrajs) [inline app/javascript/dashboard/api/auth.js:145]: Why is there an explicit check? Can't we do !profileAttributes[key]?

Comment (nithindavid) [reply app/javascript/dashboard/api/auth.js:145]: Boolean values like `false` will be ignored if we don't do explicit check

**Comment by pranavrajs** [comment]
@nithindavid A couple of comments.

1. If I enable the signature in my profile, why do I need to enable it again when I reply to a message? 
2. The terms message signature and email signature are used interchangeably. Please stick to the "message signature".
3. The UI needs to be better, this one takes half of the existing space to just show the signature which cannot be edited. Can we utilize the space below as shown in the image?

Screenshot 2022-02-07 at 10 21 55 PM

**Comment by nithindavid** [comment]
@pranavrajs These suggestions make sense. I have changed them except the space below sign.

It comes from the `p` tag in sign since we are using rich text editor we shouldn't be removing the margin as it'll be rendered weird  if there are multiple p tags.

File: app/javascript/dashboard/i18n/locale/en/settings.json [hunk 3]
```diff
@@ -19,6 +19,18 @@
...
       },
+      "MESSAGE_SIGNATURE_SECTION": {
+        "TITLE": "Personal message signature",
+        "NOTE": "Having a personal message signature will append it all the outgoing messages automatically. You can paste rich text signature also here.",
+        "BTN_TEXT": "Save message signature",
+        "API_ERROR":"Couldn't save signature! Try again",
+        "API_SUCCESS": "Signature saved successfully"
+      },
+      "MESSAGE_SIGNATURE": {
+        "LABEL": "message signature",
```
Comment (pranavrajs) [inline app/javascript/dashboard/i18n/locale/en/settings.json:30]: Message Signature

Comment (pranavrajs) [inline app/javascript/dashboard/i18n/locale/en/settings.json:30]:
```suggestion
        "LABEL": "Message Signature",
```

File: app/javascript/dashboard/i18n/locale/en/settings.json [hunk 4]
```diff
@@ -19,6 +19,18 @@
...
+        "ERROR": "message signature cannot be empty",
```
Comment (pranavrajs) [inline app/javascript/dashboard/i18n/locale/en/settings.json:31]:
```suggestion
        "ERROR": "Message Signature cannot be empty",
```

File: app/javascript/dashboard/i18n/locale/en/settings.json [hunk 5]
```diff
@@ -19,6 +19,18 @@
         "TITLE": "Profile",
         "NOTE": "Your email address is your identity and is used to log in."
       },
+      "MESSAGE_SIGNATURE_SECTION": {
+        "TITLE": "Personal message signature",
+        "NOTE": "Having a personal message signature will append it all the outgoing messages automatically. You can paste rich text signature also here.",
```
Comment (pranavrajs) [inline app/javascript/dashboard/i18n/locale/en/settings.json:24]:
```suggestion
        "NOTE": "Create a personal message signature that would be added to all the messages you send from the platform. Use the rich content editor to create a highly personalised signature.",
```

**Comment by pranavrajs** [comment]
@nithindavid For some reason the signature button is not shown in my options. 
Screenshot 2022-02-08 at 11 52 57 AM

**Comment by nithindavid** [comment]
@pranavrajs It's only enabled on email channel as that was our expectation initially also as we are using richtext signatures. We could enable for livechat also.

**Pull request merged**